manifests/params.pp

```diff
@@ -186,13 +186,16 @@
   $tftp_syslinux_files = undef
 
   # DHCP settings - requires optional DHCP puppet module
-  $dhcp                   = false
-  $dhcp_listen_on         = 'https'
-  $dhcp_managed           = true
-  $dhcp_interface         = 'eth0'
-  $dhcp_gateway           = '192.168.100.1'
-  $dhcp_range             = false
-  $dhcp_option_domain     = [$::domain]
+  $dhcp                    = false
+  $dhcp_split_config_files = true # smart-proxy 1.11+
+  $dhcp_listen_on          = 'https'
+  $dhcp_managed            = true
+  $dhcp_provider           = 'isc'
```

Comment:
At some point we could introduce a parameter ```dhcp_provider_prefix```, but that's ugly. I agree we could have been wrong to add the prefix to ```dns_provider```. It was added in https://github.com/theforeman/puppet-foreman_proxy/pull/197.